File: C++/Programy/Horner.cpp
#include <iostream>
using namespace std;

void drukujw(int n, float tbwsp[]) {
    int i = 0;
    for (i = 0; i < n; i++) {
        cout << tbwsp[i] << "x^" << n-i << " + ";
    }
    cout << tbwsp[i] << endl;

}

float horner_it(int n, float tbwsp[], float x)
{
    float wynik = tbwsp[0];
    for (int i = 1; i < n + 1; i++)
    { 
        wynik = wynik * x + tbwsp[i];
      
    }
    return wynik;

}
int main(int argc, char **argv)
{
    int n = 0; 
    float x = 0;
    cout << "Stopien: ";
    cin >> n;
    float tbwsp[n + 1];
    for (int i = 0; i <= n; i++) {
        cout << "Wspolczynnik przy potedze " << i+1 << ": ";
        cin >> tbwsp[i];
    }
    cout << "Argument: ";
    cin >> x;

    drukujw(n, tbwsp);
    cout << "Wynosi: " << horner_it(n, tbwsp, x) << endl;
    return 0;
}

File: Python/Programy/horner.py
def drukujw(n,tbwsp):
	for i in range(n):
		print(f"{tbwsp[i]}x^{n-i} + ", end="")
	print(tbwsp[n])


def horner_it(n, tbwsp ,x):
	wynik = tbwsp[0];

	for i in range(1, n+1):
		wynik = wynik * x + tbwsp[i];
	return wynik;
	


def main(args):
	
	

	n=int(input("Jaki stopień wielomianu: "))
	tbwsp = []

	for i in range(n+1):
		tbwsp.append(float(input(f"Współczynnik przy potędze {n-i}: ")))
	x=float(input("Jaki argument x: "))

	print("Wartość wielomianu o postaci: ") 
	drukujw(n, tbwsp)
	print(f"wynosi: {horner_it(n,tbwsp,x)}")
    
	return 0
	
	
if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
